src/Adapter/Order/OrderAmountUpdater.php

```diff
@@ -315,17 +315,15 @@ private function updateOrderTotals(Order $order, Cart $cart, int $computingPreci
                 $order->total_discounts = $order->total_discounts - $order->total_shipping_tax_incl + $totalShippingTaxIncluded;
                 $order->total_discounts_tax_excl = $order->total_discounts_tax_excl - $order->total_shipping_tax_excl + $totalShippingTaxExcluded;
                 $order->total_discounts_tax_incl = $order->total_discounts_tax_incl - $order->total_shipping_tax_incl + $totalShippingTaxIncluded;
+            } else {
+                $order->total_paid -= ($order->total_shipping_tax_incl - $totalShippingTaxIncluded);
```

Comment:
this operation is weird for me
Whatever the shipping value is, this will result in 

$order->total_paid -= 0
$order->total_paid_tax_incl -= 0
$order->total_paid_tax_excl -= 0

Reply:
why not test that case earlier and Do nothing

Reply:
oh, I see in the case where the carrier is updated it will result removing 0 from shipping. But in the case the products are updated, th eprice won't be the same. My bad

Reply:
Because `$cart->getOrderTotal(true, CART::ONLY_BOTH...) calculate with shipping... So if we change carrier, `$cart->getOrderTotal(true, CART::ONLY_BOTH...)` take the shipping for the new carrier, we need to substract it and add the old one.

Reply:
oh, ok I get it, it's because total_paid is recalculated l284, so with the new shipping price. This operation is only for resetting the former shipping price. Sorry for bothering you with that. thanks